### PR TITLE
SAM-2242 fetch pool sizes in one query instead of in multiple queries

### PR DESCRIPTION
**Pull request by ottenhoff**
https://jira.sakaiproject.org/browse/SAM-2242

So if a user has 1000 pools ..... old behavior would result in one thousand queries instead of one big query

### CODE REVIEW

**Comment by ern** [comment]
Another samigo db improvement +1 for merge.

**Comment by axxter99** [comment]
+1

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/QuestionPoolFacadeQueries.java [hunk 1]
```diff
@@ -188,28 +184,28 @@ public QuestionPoolIteratorFacade getAllPools(String agentId) {
 
   public QuestionPoolIteratorFacade getAllPoolsWithAccess(String agentId) {
 	  ArrayList qpList = new ArrayList();
+	  HashMap counts = new HashMap<Long, Integer>();
```
Comment (buckett) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/QuestionPoolFacadeQueries.java]: Generics on the declaration?

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/QuestionPoolFacadeQueries.java [hunk 2]
```diff
@@ -188,28 +184,28 @@ public QuestionPoolIteratorFacade getAllPools(String agentId) {
...
-			  // I really wish we don't need to populate  the questionpool size & subpool size for JSF
-			  // watch this for performance. hope Hibernate is smart enough not to load the entire question
-			  // - daisy, 10/04/04
-
-			  // The comment below is to not recover all questions of items
-			  //populateQuestionPoolItemDatas(qpp);
-
-			  // I do this call, after it did in populateQuestionPoolItemData, to recover the number of subpools that will be show in the root of pools.
-			  qpp.setSubPoolSize( Integer.valueOf(getSubPoolSize(qpp.getQuestionPoolId())));
+			  qpp.setSubPoolSize( (int) counts.get(qpp.getQuestionPoolId()));
```
Comment (buckett) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/QuestionPoolFacadeQueries.java]: Can drop the cast if you add the generics.

**Comment by buckett** [comment]
Looks nice Sam. Comments aren't meant as show stoppers, just feedback.

**Comment by ottenhoff** [comment]
Okay, changes made per @buckett

**Comment by buckett** [comment]
Magic, and a rebase, nice. :cake:

**Pull request merged**